ikarus/io/resultevaluators.hh

```diff
@@ -26,40 +27,76 @@ struct VonMises
 {
   /**
    * \brief Calculate the result quantity (von Mises stress)
-   * \param resultArray EigenMatrix containing the stress state
+   * \param resultArray EigenVector containing the stress state in Voigt notation
    * \param comp component of result (not used here)
    * \tparam R Type of the matrix
    * \return von Mises stress
    */
   template <typename R>
   double operator()(const R& resultArray, [[maybe_unused]] const int comp) const {
-    const auto s_x = resultArray(0, 0);
-    const auto s_y = resultArray(1, 0);
+    const auto s_x = resultArray[0];
```

Comment:
I would just write here:

```cpp
auto sigma = fromVoigt(resultArray,false);

auto I2= 1.0/2.0*(sigma.squaredNorm()-1.0/3.0*Dune::power(sigma.trace(),2)) ;
return std::sqrt(3.0*I2);

```
which should work for 3D and 2D plane stress, see attachement
[VonMises.mw.txt](https://github.com/user-attachments/files/18436801/VonMises.mw.txt)